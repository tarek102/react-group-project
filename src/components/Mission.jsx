import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Table } from 'react-bootstrap';
import { fetchMissions } from '../redux/mission/mission';
import AllMission from './AllMission';

const Missions = () => {
  const dispatch = useDispatch();
  const missionsData = useSelector((state) => state.missions);

  useEffect(() => {
    if (missionsData.length < 1) {
      dispatch(fetchMissions());
    }
  }, []);

  return (
    <>
      <div className="container mt-5">
        <div className="table-responsive">
          <Table striped bordered>
            <thead>
              <tr>
                <th scope="col">Mission</th>
                <th scope="col">Description</th>
                <th scope="col">Status</th>
                <th scope="col">.</th>
              </tr>
            </thead>
            <tbody>
              {
                missionsData.map((item) => (
                  <AllMission
                    id={item.mission_id}
                    name={item.mission_name}
                    description={item.description}
                    reserved={item.reserved}
                    key={item.mission_id}
                  />
                ))

              }
            </tbody>
          </Table>
        </div>

      </div>

    </>
  );
};

export default Missions;
