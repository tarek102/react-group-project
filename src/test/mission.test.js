import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Mission from '../components/Mission';

it('missions display correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Mission />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});