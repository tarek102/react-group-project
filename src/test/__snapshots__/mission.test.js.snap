// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`missions display correctly 1`] = `
<div
  className="container mt-5"
>
  <div
    className="table-responsive"
  >
    <table
      className="table table-striped table-bordered"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Mission
          </th>
          <th
            scope="col"
          >
            Description
          </th>
          <th
            scope="col"
          >
            Status
          </th>
          <th
            scope="col"
          >
            .
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
